bundles/model/org.eclipse.smarthome.model.sitemap/src/org/eclipse/smarthome/model/validation/SitemapValidator.xtend

```diff
@@ -27,92 +27,84 @@ import java.math.BigDecimal
+                    SitemapPackage.Literals.SITEMAP.getEStructuralFeature(SitemapPackage.SITEMAP__NAME));
+                return
+            }
+        }
+    }
 
-	@Check
-	def void checkFramesInWidgetList(LinkableWidget widget) {
-		if (widget instanceof Frame) {
+    @Check
+    def void checkFramesInWidgetList(LinkableWidget widget) {
+        if (widget instanceof Frame) {
 
-			// we have a dedicated check for frames in place
-			return;
-		}
-		var	containsFrames = false
-		var	containsOtherWidgets = false
-		for (Widget w : widget.children) {
-			if (w instanceof Frame) {
-				containsFrames = true
-			} else {
-				containsOtherWidgets = true
-			}
-			if (containsFrames && containsOtherWidgets) {
-				error("Linkable widget should contain either only frames or none at all",
-					SitemapPackage.Literals.FRAME.getEStructuralFeature(SitemapPackage.LINKABLE_WIDGET__CHILDREN));
-				return
-			}
-		}
-	}
+            // we have a dedicated check for frames in place
+            return;
+        }
+        var containsFrames = false
+        var containsOtherWidgets = false
+        for (Widget w : widget.children) {
+            if (w instanceof Frame) {
+                containsFrames = true
+            } else {
+                containsOtherWidgets = true
+            }
+            if (containsFrames && containsOtherWidgets) {
+                error("Linkable widget should contain either only frames or none at all",
+                    SitemapPackage.Literals.FRAME.getEStructuralFeature(SitemapPackage.LINKABLE_WIDGET__CHILDREN));
+                return
+            }
+        }
+    }
 
-	@Check
-	def void checkSetpoints(Setpoint sp) {
-	    if(sp.item === null)
-	    {
-	        error("Setpoint without an item defined",
-                    SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT));
-            return
-	    }
+    @Check
+    def void checkSetpoints(Setpoint sp) {
+        if (sp.item === null) {
+            error("Setpoint without an item defined",
+                SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT));
+        }
 
-	    if(BigDecimal.ZERO.equals(sp.step))
-	    {
-	        error("Setpoint on item '" +sp.item+ "' has step size of 0",
-                    SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT__STEP));
-	        return
-	    }
+        if (BigDecimal.ZERO == sp.step) {
+            error("Setpoint on item '" + sp.item + "' has step size of 0",
+                SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT__STEP));
+        }
 
-	    if(sp.step < BigDecimal.ZERO)
-        {
-            error("Setpoint on item '" +sp.item+ "' has negative step size",
-                    SitemapPackage.Literals.SETPOINT.getEStructuralFeature(SitemapPackage.SETPOINT__STEP));
-            return
+        if (sp.step < BigDecimal.ZERO) {
```

Comment:
Just played around with it a little: This and the below comparison fail with an NPE is the values are not explicitly set.